src/odemis/gui/cont/streams.py

```diff
@@ -432,6 +446,32 @@ def _on_stream_peak(self, evt):
         else:
             logging.error("No peak method corresponding to state %s", evt.state)
 
+    def _on_z_index(self, zindex):
+        
+        self.tab_data_model.zpos.unsubscribe(self._on_z_pos)
+
+        metadata = self.stream.getRawMetadata()
+        zstart = metadata[model.MD_POS][2]
```

Comment:
It's not `zstart`, it's `zcenter`. => You need to also take into account the zindex.range.